cpp/src/arrow/table.cc

```diff
@@ -532,6 +550,19 @@ bool Table::Equals(const Table& other) const {
   return true;
 }
 
+Status Table::Compact(MemoryPool* pool, std::shared_ptr<Table>* out) const {
+  const int ncolumns = num_columns();
+  std::vector<std::shared_ptr<Column>> compacted_columns;
+  compacted_columns.reserve(ncolumns);
```

Comment:
This line can be removed, you can pass the hint to the previous declaration (see https://en.cppreference.com/w/cpp/container/vector/vector).

Reply:
Isn't there a small semantic difference?  I thought the constructor version allocates and construct the default element.  Reserve just allocates (and doesn't affect the logical size of the vector).

Reply:
The original code does reserve() and push_back() while the new revision uses the n-element constructor and indices into the vector.

The new revision is fewer lines of code though. (the generated binary is also different but we are not at the point where the difference matters, I think).